src/hardware/signal_generator.rs

```diff
@@ -20,7 +20,7 @@ pub enum Signal {
 /// Where `<signal>` may be any of [Signal] variants, `frequency` specifies the signal frequency
 /// in Hertz, `symmetry` specifies the normalized signal symmetry which ranges from 0 - 1.0, and
 /// `amplitude` specifies the signal amplitude in Volts.
-#[derive(Copy, Clone, Debug, Miniconf, Deserialize)]
+#[derive(Copy, Clone, Debug, Miniconf, Serialize, Deserialize)]
```

Comment:
I don't think we should need `Serialize` or `Deserialize` on this structure anymore, since it is never configured as a single value, correct?

Reply:
Ack. But it doesn't hurt.

Reply:
Agreed that it doesn't hurt. Just trying to validate my understanding of the current state of `miniconf` :)

Reply:
Yes. One early annoyance I hit is that I can't make a `Miniconf` struct `Serialize` or `Deserialize` if it contains a `miniconf::Array`. Not that it's needed urgently right now, but it does show the impact of those newtypes.